code/game/objects/items/crayons.dm

```diff
@@ -380,6 +380,9 @@
 	if(edible && (M == user))
 		user << "You take a bite of the [src.name]. Delicious!"
 		var/eaten = use_charges(5)
+		if(!reagents.total_volume) //Prevents divsion by zero
+			check_empty()
```

Comment:
What I came across is that without the check_empty call you could eat the crayon an infinite amount of times and it would only be deleted if you attempted to draw something on the floor with it.